stripe/src/main/java/com/stripe/android/Stripe.kt

```diff
@@ -1653,6 +1697,16 @@ class Stripe internal constructor(
         )
     }
 
+    /**
+     * Get a [LifecycleCoroutineScope] if available. Otherwise, default to [MainScope].
+     */
+    private fun getLifecycleScope(
+        activity: Activity
+    ): CoroutineScope = when (activity) {
+        is ComponentActivity -> activity.lifecycleScope
+        else -> MainScope()
```

Comment:
Yeah, this isn't ideal and mainly an attempt to make things better for users using `ComponentActivity` subclasses without introducing a breaking API change (will consider doing that in a future PR).

It might actually make sense to use `Dispatchers.IO` as the fallback, which is the existing behavior.